src/server/scripts/Kalimdor/Firelands/boss_baleroc.cpp

```diff
@@ -0,0 +1,1257 @@
+ * option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "ScriptMgr.h"
+#include "ScriptedCreature.h"
+#include "PassiveAI.h"
+#include "SpellScript.h"
+#include "SpellAuraEffects.h"
+#include "firelands.h"
+
+enum Spells
+{
+    // Baleroc Trash
+    SPELL_FLAME_TORRENT             = 100795,
+    SPELL_FIERY_TORMENT             = 100797,
+    SPELL_FIERY_TORMENT_DAMAGE      = 100802,
+    SPELL_EARTHQUAKE                = 100724,
+    SPELL_MAGMA_CONDUIT             = 100728,
+    SPELL_ERUPTION                  = 100755,
+    SPELL_SUMMON_MAGMAKIN           = 100746,
+
+    // Baleroc
+    //SPELL_LEASH                   = 101514, // Server-side, used to keep him in his encounter area? TrinityCore does not need a spell to handle this
+    SPELL_BLADES_OF_BALEROC         = 99342,
+    SPELL_INFERNO_BLADE             = 99350,
+    SPELL_INFERNO_STRIKE            = 99351,
+    SPELL_DECIMATION_BLADE          = 99352,
+    SPELL_DECIMATION_BLADE_2        = 99405,
+    SPELL_DECIMATING_STRIKE         = 99353,
+    SPELL_BLAZE_OF_GLORY            = 99252,
+    SPELL_INCENDIARY_SOUL           = 99369,
+    SPELL_SHARDS_OF_TORMENT         = 99259,
+    SPELL_SHARDS_OF_TORMENT_SUMMON  = 99260,
+    SPELL_TORMENT_PRE_VISUAL        = 99258,
+    SPELL_TORMENT_ACTIVE            = 99254,
+    SPELL_TORMENT_PERIODIC          = 99255,
+    SPELL_WAVE_OF_TORMENT           = 99261,
+    SPELL_TORMENTED                 = 99257,
+    SPELL_TORMENT                   = 99256,
+    SPELL_COUNTDOWN                 = 99515,
+    SPELL_COUNTDOWN_AURA            = 99516,
+    SPELL_COUNTDOWN_3               = 99517,
+    SPELL_COUNTDOWN_AOE_EXPLOSION   = 99518,
+    SPELL_COUNTDOWN_VISUAL_LINK     = 99519,
+    SPELL_VITAL_SPARK               = 99262,
+    SPELL_VITAL_FLAME               = 99263,
+    SPELL_BERSERK                   = 26662,
+};
+
+enum Emotes
+{
+    SAY_AGGRO                       = 0,
+    SAY_SHARDS_OF_TORMENT           = 1,
+    SAY_INFERNO_BLADE               = 2,
+    SAY_DECIMATION_BLADE            = 3,
+    SAY_KILL                        = 4,
+    SAY_DEATH                       = 5,
+    SAY_ENRAGE                      = 6,
+    EMOTE_ENRAGE                    = 7,
+    EMOTE_DECIMATION_BLADE          = 8,
+    EMOTE_INFERNO_BLADE             = 9,
+};
+
+enum Guids
+{
+    GUID_TORMENTED                  = 1,
+};
+
+enum Actions
+{
+    ACTION_EQUIP_DEFAULT            = 1,
+    ACTION_EQUIP_INFERNO_BLADE      = 2,
+    ACTION_EQUIP_DECIMATION_BLADE   = 3,
+};
+
+
+enum Misc
+{
+    EQUIP_DEFAULT                   = 1,
+    EQUIP_INFERNO_BLADE             = 2,
```

Comment:
Does 71138 + 71082 entries cause Baleroc's weapon to shine when he empowers his blades ?
If not, use these entries:

```
enum ItemEntries
{
    NORMAL_BLADE_ENTRY        = 71055,
    INFERNO_BLADE_ENTRY       = 94155,
    DECIMATION_BLADE_ENTRY    = 94157,
};
```